src/site/_includes/components/cookie/__tests__/cookie-settings-page.test.ts

```diff
@@ -0,0 +1,62 @@
+import { fireEvent, getByRole } from '@testing-library/dom'
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
+import { initCookieSettingPage } from '..'
+import * as analytics from '../analytics'
+import { COOKIE_PREFERENCE_KEY, deleteCookie, setCookie } from '../cookie'
+
+const html = `
+<div class="js-cookie-notification-banner" hidden>Success</div>
+<form  class="js-cookie-setting-form" name="cookie-settings">
+  <label for="analytics-cookies-yes">Yes</label>
+  <input type="radio" id="analytics-cookies-yes" name="analytics-cookies" value="1" />
+  <label for="analytics-cookies-no">No</label>
+  <input type="radio" id="analytics-cookies-no" name="analytics-cookies" value="0" />
+  <button type="submit">Submit</button>
+</form>
+`
+
+vi.mock('../cookie')
+
+const FormDataMock = vi.fn((form: HTMLFormElement) => ({
+  get: vi.fn((key: string) => (
+    [...form.querySelectorAll<HTMLInputElement>(`[name="${key}"]`)].find((input) => input.checked)?.value
+  )),
+}))
+
+describe('initCookieSettingForm', () => {
+  beforeEach(() => {
+    document.body.innerHTML = html
+    window.googleTagId = 'G-1234'
+    vi.stubGlobal('FormData', FormDataMock)
+  })
+
+  afterEach(() => {
+    vi.resetModules()
+    vi.unstubAllGlobals()
+    document.cookie = 'design-system-cookie-preference=unset;expires=Thu, 01 Jan 1970 00:00:00 GMT;path=/'
+  })
+
+  it('submitting yes calls analyticSetting with true', async () => {
+    const analyticsSettingSpy = vi.spyOn(analytics, 'analyticsSetting')
+    initCookieSettingPage()
+
+    getByRole<HTMLInputElement>(document.body, 'radio', { name: 'Yes' }).checked = true
+    fireEvent(getByRole(document.body, 'form'), new Event('submit', { bubbles: true }))
```

Comment:
happy-dom don't support click event on checkbox and radios at the moment so I had to assign it also submit event are not triggered but by clicking `<button type="submit"/>